services/spar/src/Spar/Types.hs

```diff
@@ -311,3 +315,32 @@ instance FromJSON SsoSettings where
 instance ToJSON SsoSettings where
   toJSON SsoSettings {defaultSsoCode} =
     object ["default_sso_code" .= defaultSsoCode]
+
+-- | Used as a lookup key for 'UnqualifiedNameID' that only depends on the
+-- lowercase version of the identifier. Use 'normalizeUnqualifiedNameId' or
+-- 'normalizeQualifiedNameId' to create values.
+newtype NormalizedUNameID = NormalizedUNameID {unNormalizedUNameID :: Text}
+  deriving stock (Eq, Ord, Generic)
+
+instance Cql NormalizedUNameID where
+  ctype = Tagged TextColumn
+  toCql = CqlText . unNormalizedUNameID
+  fromCql (CqlText t) = pure $ NormalizedUNameID t
+  fromCql _ = Left "NormalizedNameID: expected CqlText"
+
+normalizeUnqualifiedNameId :: SAML.UnqualifiedNameID -> NormalizedUNameID
+normalizeUnqualifiedNameId = NormalizedUNameID . foldCase . nameIdTxt
```

Comment:
Maybe I'm misunderstanding something. But I see two different cases I'd distinguish: new accounts, existing accounts.
For new accounts we can simply refuse to set up accounts that would clash and tell people that the email can't be used. I'd treat all identifiers as insensitive and handle them the same way.
For existing accounts we have to go through and make sure we don't have clashing emails/ids for any accounts. Then the only remaining question would be how to deal with existing issues.
This way I don't think there should be any issues with case confusion/collisions in the system.

Reply:
The problem is SAML-autoprovisioning: if there is no SCIM and the SAML credentials aren't found in cassandra, users are created spontaneously.  If a team uses this setup, there is no way to distinguish between credentials that should create a new account and those that should find one that already exists.

Maybe we should consider discontinuing SAML-only teams, but we already have some of those in prod, and it'll be a painful transition.

Reply:
Maybe we can live with the risk for those teams, and prevent new teams from using this setup?

Reply:
Ok, then it sounds like something we could just put in the docs, tell people to be aware, and take the risk.

Reply:
> For existing accounts we have to go through and make sure we don't have clashing emails/ids for any accounts

There are only 4 clashes for existing accounts on prod, all of which are not a real problem (for details see PR description). For new accounts we need to document that qualifiers and casing is ignored. As for security risks: All NameIds are always qualified by the Idp issuer url, so any clashes that might happen are namespaced by the Idp.

---

> I'm leaning towards "ok to ignore email case, but the other distinctions should not be necessary". 

I'd like to err on the side of more equivalence of identifiers than more potential practical problems. Let's assume we don't ignore the format, e.g. `unspecified` vs. `email`. Then there is following practical problem:

1. Admin configures a SSO login with `unspecified` NameId, however the identifiers are email addresses. This is a slight misconfiguration, but one that is allowed and works. This happends in practice.
2. Hundreds of users get autoprovisioned that way 
3. Admin creates SCIM token. However since the identifiers are email addresses SCIM creates NameId with `email` format. The users created with SCIM are not able to log in.
4. If the admin changes the Idp to send the `email` format then all users created in step 2 are locked out of their accounts. 

There are other scenarios that lead to the same outcome: Some of our admin users can't configure their IdP to omit the `NameQualifier` for example, so I think. This always creates mismatches with SCIM that can only be resolved by reconfiguring the Idp.